crates/script/src/transaction.rs

```diff
@@ -1,240 +1,180 @@
 use super::ScriptResult;
 use alloy_dyn_abi::JsonAbiExt;
-use alloy_primitives::{hex, Address, Bytes, TxKind, B256};
+use alloy_primitives::{hex, Address, TxKind, B256};
 use eyre::{Result, WrapErr};
+use forge_script_sequence::TransactionWithMetadata;
 use foundry_common::{fmt::format_token_raw, ContractData, TransactionMaybeSigned, SELECTOR_LEN};
 use foundry_evm::{constants::DEFAULT_CREATE2_DEPLOYER, traces::CallTraceDecoder};
 use itertools::Itertools;
 use revm_inspectors::tracing::types::CallKind;
-use serde::{Deserialize, Serialize};
 use std::collections::BTreeMap;
 
-#[derive(Clone, Debug, Default, Serialize, Deserialize)]
-#[serde(rename_all = "camelCase")]
-pub struct AdditionalContract {
-    #[serde(rename = "transactionType")]
-    pub opcode: CallKind,
-    pub address: Address,
-    pub init_code: Bytes,
+#[derive(Debug)]
+pub struct ScriptTransactionBuilder {
+    transaction: TransactionMaybeSigned,
```

Comment:
we can keep `TransactionWithMetadata` here instead, and move methods around a bit

`new` would do the same thing as `build`, leaving partially filled tx inside of builder
`set_call`,`set_create` will be builder methods filling the inner tx
`build` will just return the inner tx

a bit nitpicky but I'd like to avoid free functions here and looks like we can have a normal builder type here